pyproject.toml

```diff
@@ -18,6 +18,9 @@ python = "^3.9"
 torch = "^2.0.0"
 numpy = "~1"
 optuna = "^3.1.0"
+osqp = "^0.6.3"
+matplotlib = "^3.7.0"
+scipy = "~1"
```

Comment:
@junziz you previously had a note that `numpy==1.26` was problematic and incompatible with `osqp`. I'm not sure if you've fully adopted `poetry` for package management yet, but this configuration works for me. Please let me know if it causes problems on your end!